core/modules/wdb/QueryRunner.cc

```diff
@@ -494,13 +502,10 @@ bool QueryRunner::_dispatchChannel() {
 void QueryRunner::cancel() {
     LOGS(_log, LOG_LVL_WARN, "Trying QueryRunner::cancel() call");
     _cancelled = true;
-    // This could be called after the task has been completed, so sendChannel
-    // validation is needed.
-    auto sChannel = _task->sendChannel;
-    if (sChannel != nullptr) {
-        lock_guard<mutex> streamLock(sChannel->streamMutex);
-        sChannel->kill(streamLock);
-    }
+
+    // QueryRunner::cancel() should only be called by Task::cancel().
+    // Task::cancel() should handle cleaning up the sendChannel().
+
```

Comment:
This explanation is better to be put into the documentation section of the method in the class's header.

Reply:
QueryRunner::cancel() has comments in the header saying it should only be called by Task::cancel(). The other comment line is not completly accurate, so I removed it.